### ISSUE TITLE
Overfitting very soon

### ISSUE THREAD

**Issue by algila** [experiencor/basic-yolo-keras#179]
keras 2.1.2

I’m testing your model (training.py) on just 10 classes from COCO plus one more produced by me having in total 40.000 images. Running it on “Full Yolo” I obtain overfitting (val_loss become higher then loss) just after the first epoch (doesn't matter optimizer used) using your config.json set of parameters. 

```
       "train_times":          10,
        "pretrained_weights":   "",
        "batch_size":           16,
        "learning_rate":        1e-4,
        "nb_epoch":             60,
        "warmup_epochs":        0,

        "object_scale":         5.0 ,
        "no_object_scale":      1.0,
        "coord_scale":          1.0,
        "class_scale":          1.0,
```

I made the following tests to verify this:

trained full yolo on raccoon dataset (config file you provided) and overfit happen after 1 epoch setting warmup_epochs to 0 or to 3

```
Full Yolo "warmup_epochs":        3,
Epoch 1/3
 99/100 [============================>.] - ETA: 0s - loss: 0.1263Epoch 00001: val_loss improved from inf to 0.08456, saving model to full_yolo_prova2.h5
100/100 [==============================] - 47s 473ms/step - loss: 0.1254 - val_loss: 0.0846
Epoch 2/3
 99/100 [============================>.] - ETA: 0s - loss: 0.0262Epoch 00002: val_loss improved from 0.08456 to 0.02017, saving model to full_yolo_prova2.h5
100/100 [==============================] - 45s 448ms/step - loss: 0.0261 - val_loss: 0.0202
Epoch 3/3
 99/100 [============================>.] - ETA: 0s - loss: 0.0098Epoch 00003: val_loss improved from 0.02017 to 0.01159, saving model to full_yolo_prova2.h5
100/100 [==============================] - 45s 449ms/step - loss: 0.0097 - val_loss: 0.0116

Full Yolo "warmup_epochs":        0,
Epoch 1/30
 99/100 [============================>.] - ETA: 0s - loss: 0.6624Epoch 00001: val_loss improved from inf to 0.46435, saving model to full_yolo_prova2.h5
100/100 [==============================] - 48s 481ms/step - loss: 0.6595 - val_loss: 0.4644
Epoch 2/30
 99/100 [============================>.] - ETA: 0s - loss: 0.2967Epoch 00002: val_loss did not improve
100/100 [==============================] - 44s 436ms/step - loss: 0.2978 - val_loss: 0.4644
Epoch 3/30
 99/100 [============================>.] - ETA: 0s - loss: 0.2255Epoch 00003: val_loss improved from 0.46435 to 0.41080, saving model to full_yolo_prova2.h5
100/100 [==============================] - 46s 457ms/step - loss: 0.2250 - val_loss: 0.4108
```
I tested also Tiny yolo the same:

```
Epoch 1/30
 99/100 [============================>.] - ETA: 0s - loss: 2.9454Epoch 00001: val_loss improved from inf to 1.74970, saving model to full_yolo_prova2.h5
100/100 [==============================] - 21s 213ms/step - loss: 2.9274 - val_loss: 1.7497
Epoch 2/30
 99/100 [============================>.] - ETA: 0s - loss: 1.1960Epoch 00002: val_loss did not improve
100/100 [==============================] - 19s 195ms/step - loss: 1.1937 - val_loss: 2.4162
Epoch 3/30
 99/100 [============================>.] - ETA: 0s - loss: 1.0506Epoch 00003: val_loss improved from 1.74970 to 1.52344, saving model to full_yolo_prova2.h5
100/100 [==============================] - 21s 211ms/step - loss: 1.0527 - val_loss: 1.5234
```
any ideas how to solve ? Did you experienced the same ?

**Comment by Sayantan17** [comment]
How you created your own dataset with the xml format of .json Format used for annotations in any random config.json file? Can you guide me?

**Comment by algila** [comment]
I took photos with my mobile and then used [https://github.com/tzutalin/labelImg] to annotate in PASCAL VOC

**Comment by experiencor** [comment]
@algila Yes, my training progresses similarly to yours. But did the network perform visually okay?

**Comment by Sayantan17** [comment]
Oh thanks I will try & tell if there's any problem @algila 😊

**Comment by Sayantan17** [comment]
@algila  shall I run the python code labelImg code to convert image dataset which like u gathered from mobile to.VOC format (xml) ?

**Comment by zeynali** [comment]
@Sayantan17 , yes , you can create xml annotation files for your photos , you first run the labelimg and then draw bounding box for objects in each photos and put each of them label , then save them ans so on .
